### ISSUE TITLE
As a user, I want the WRES to output a sqlite artifact for my evaluations

### ISSUE THREAD

**Issue by epag** [NOAA-OWP/wres#260]
---

Author Name: **Chris** (Chris)
Original Redmine Issue: 92406, https://vlab.noaa.gov/redmine/issues/92406
Original Date: 2021-05-25

---

I was just looking at a ticket for the new @csv2@ schema and it struck me that the change could potentially be breaking if working with CSV since it's a semi-structured product, but that wouldn't really be a problem at all if it were output in a structured output, like a sqlite blob. A portable master blob would also be an incredibly helpful tool after the fact considering that many tools (like python + pandas) can read and manipulate it no problem.

spatialite would be preferable, but not always doable for generation. My laptop can read spatialite no problem, for example, but can't easily write it without recompiling stuff like python to look for a different sqlite3 executable (MUCH easier said than done).

I put COWRES in the subject since _something_ outputting the sqlite would be neat - it just doesn't necessarily need to be the core WRES - and I'm setting this as low priority since it would be handy, but no one is actively clamoring for it.

**Comment by epag** [comment]
---

Original Redmine Comment
Author Name: **James** (James)
Original Date: 2024-02-01T13:13:31Z

---

Increasing the priority of this ticket. If we (continue to) accept that wres is the wrong tool for fancy graphics and custom evaluation products, then it becomes super important to facilitate the easy slicing/dicing/customization of our statistics datasets.

**Comment by epag** [comment]
---

Original Redmine Comment
Author Name: **Hank** (Hank)
Original Date: 2024-02-29T14:07:21Z

---

From Chris via chat:

> I didn't know it at the time of writing, but geopackage is a type of sqlite database; imagine postgis, but for sqlite. THAT would be a great artifact if you could ship it with geometry because arc and qgis love geopackage.

Hank

**Comment by epag** [comment]
---

Original Redmine Comment
Author Name: **James** (James)
Original Date: 2024-02-29T14:09:34Z

---

Yes, good call, that's what we should aim for.

**Comment by HankHerr-NOAA** [comment]
Chris, once again, mentioned the idea of using GeoPackage:

https://www.geopackage.org/

James already bumped the priority on this once while it was in VLab. I wonder if we should bump it now that its in github. Implementing that might make the use of WRES more attractive to Python users. 

Hank

**Comment by HankHerr-NOAA** [comment]
Agreed. We do have active Python users, in that they are using Python script to run evaluations, but none of them have asked for this feature, so its not a barrier to use. Then again, I'm not sure how many of them know to ask for this feature.

Anyway, its on our radar,

Hank

**Comment by james-d-brown** [comment]
What I mean by Python users, specifically, in this context is a user that is importing (or wanting to import) the statistics from WRES into a Python workflow (e.g., for visualization), not so much how they manage service requests and responses, for which the language is incidental. Do we have any users like that?

**Comment by james-d-brown** [comment]
Thanks!

In that case, I think we should keep this as a higher priority, but lower than high priority bug fixes and user requests. It's something we can return to when we've exhausted those first two categories, which obviously does happen periodically.
